### PR TITLE
1851 mjrc split

### PR DESCRIPTION
**Pull request by AchyutRatkanthiwar**
## Description
Trello ticket [#1851](https://trello.com/c/3bwRJVfY/1851-pivot-separate-mergejobrolecolumns-into-two-different-functions-as-prep-for-refactor)

Split the shared `merge_job_role_columns` function in Polars Utils into two independent copies, one per calling pipeline: `merge_legacy_job_role_columns` in the ASCWDS ingest clean job, and `reduce_to_published_roles` in the SLV prepare job. This removes the shared dependency ahead of a future refactor of the SLV pipeline's job role handling.

## Testing
- [x] Unit tests passing
- [x] Successful [SLV Step Function run](https://eu-west-2.console.aws.amazon.com/states/home?region=eu-west-2#/v2/executions/details/arn:aws:states:eu-west-2:856699698263:execution:1851-mjrc-split-Ind-CQC-SLV:c75c6428-84a8-4ed6-9fef-b97da2645167)
- [x] Successful [Transform ASCWDS Data Step Function run](https://eu-west-2.console.aws.amazon.com/states/home?region=eu-west-2#/v2/executions/details/arn:aws:states:eu-west-2:856699698263:execution:1851-mjrc-split-Transform-ASCWDS-Data:858b95cc-ad0b-4ade-9fd7-20e4ad3330fa)
- [x] Outputs checked in Athena

## Checklist (delete if not relevant)
- [x] Unit tests added/amended
- [x] Docstrings added/updated
- [x] Updated CHANGELOG
- [x] Code reviewed by AI
- [x] Moved Trello ticket to PR column

## Reviewer checklist
- [ ] PR solves the Trello ticket requirements
- [ ] Outputs appear reasonable and understandable
- [ ] The overall approach is appropriate and not over-engineered
- [ ] No obvious scalability, performance or interdependency concerns
- [ ] Tests appropriately cover the main behaviour/edge cases
- [ ] Naming and structure are broadly understandable and consistent
- [ ] Docstrings are sufficient for future maintenance
- [ ] Docstrings cover non-obvious behaviour

### CODE REVIEW

File: projects/_07_workforce_characteristics/_01_starters_leavers_vacancies/tests/fargate/utils/test_prepare_utils.py [hunk 1]
```diff
@@ -1,11 +1,146 @@
...
+
 import projects._07_workforce_characteristics._01_starters_leavers_vacancies.fargate.utils.prepare_utils as job
+from utils.column_names.cleaned_data_files.ascwds_workplace_cleaned import (
+    AscwdsWorkplaceCleanedColumns as AWPClean,
+)
 
 PATCH_PATH = "projects._07_workforce_characteristics._01_starters_leavers_vacancies.fargate.utils.prepare_utils"
 
 
+@dataclass
```
Comment (MarionJHolloway) [inline projects/_07_workforce_characteristics/_01_starters_leavers_vacancies/tests/fargate/utils/test_prepare_utils.py]: These will want moving out to the test data script

File: projects/_01_ingest/ascwds/tests/fargate/utils/test_clean_workplace_utils.py [hunk 2]
```diff
@@ -245,3 +247,133 @@ def test_slv_expression_bounds_values_to_valid_range(self):
...
+@dataclass
+class MergeLegacyJobRoleColumnsTestCase:
+    id: str
+    mapping: dict[str, list[str]]
+    input_data: dict[str, Any]
+    expected_data: dict[str, Any]
+
+
+class TestMergeLegacyJobRoleColumns:
+    merge_legacy_job_role_columns_test_cases = [
```
Comment (MarionJHolloway) [inline projects/_01_ingest/ascwds/tests/fargate/utils/test_clean_workplace_utils.py]: Move test data to test data script. Not sure why they were in here in the old code structure, but probably easiest to move them now

**Review by MarionJHolloway — CHANGES_REQUESTED** [review]
Just needs the test data locations fixing. Looks like they were in the test file in the current code, but should really be in the test data script as there's quite a lot of test cases

**Review by MarionJHolloway — APPROVED** [review]
Looks good, thanks!
